### DISCUSSION TITLE
Exporting order in which records were screened

### DISCUSSION THREAD

**Discussion by MariaRapoport** [asreview/asreview#1730] [Q&A]
Hi all, 

I am piloting ASReview and wanted to see if there is any way to get the order in which records have been presented out of the application? The Export Excel does not seem to include this info to the best of my understanding. I can get the recall out of the graph on the analytics page, but that does not provide me with the order of the specific records. 

Thanks!

**Comment by PeterLombaers** [comment]
Hi @MariaRapoport !
Right now there isn't a super easy way to get that data. I have two solutions though. They both work by getting data out of the technical `.asreview` file for your project. You can get this file from the export page by choosing the `.asreview` format. Then you can get the data you want as follows:
- If you know a little bit of Python you can follow [this example](https://asreview.readthedocs.io/en/stable/example_api_asreview_file.html#State-API). The column titled `record_id` is the data you want.
- You could use the command `asreview state-inspect path/to/your/file.asreview results >> results.txt` (replace `path/to/your/file.asreview` by the location of your file). This takes the result data and puts it in a text format. Then you can open `results.txt`, with Excel for example, to get the `record_id` column.

I hope one of these two methods works for you. Let me know if you still have problems!

**Reply by JJBDub** [reply]
Peter,
When exporting all records from a simulation via the GUI, as shown in the attached screenshot, the data includes a record_id column.
Are the values different from the record_id values you get when pulling them out of the .asreview file as you describe?
![image](https://github.com/user-attachments/assets/2c015f39-19f3-4b43-a103-f820efac6c0e)


After a simulation is completed, we want to be able to trace the exact sequence of records that it went through.
When we use the exported data and order it by record_id, then calculate the cumulative inclusions (relevant records), the recall curve we obtain is not at all the one we see in the Analytics recall plot. 
In other words, when we use record_id to order the records in the "include all" dataset export, we are not getting the sequence of inclusions and exclusions (relevant and irrelevant) that is shown in the .csv export from the recall plot.

**Reply by PeterLombaers** [reply]
Hey @JJBDub! That column `record_id` in your export file is the row number of the record in your input dataset, where we start counting from 0. So if you order the data by record_id, you are just putting it back in the order of the original dataset. If you extract the data from the technical file in one of the two ways I mention above, you will get the records in the order that ASReview presented them. 

I hope this makes things more clear, if you still have some question, just let me know!

**Reply by JJBDub** [reply]
Got it! 
Thank you.

**Reply by JJBDub** [reply]
@PeterLombaers 
Your examples worked perfectly. 
We haven't been able to find a data dictionary for the tables we pulled from the .asreview file though.
Can you point us to it? 
Or else for now, we just wonder what the "label" column is.

Sorry we couldn't find it in the docs.

**Reply by Rensvandeschoot** [reply]
Is this what you are looking for: https://asreview.readthedocs.io/en/latest/progress.html#export-results?
Or via the API: https://asreview.readthedocs.io/en/latest/generated/asreview.state.SQLiteState.html

**Reply by JJBDub** [reply]
Thank you, those docs help clarify, but we still can't figure out what "label" is. It does not match the relevant/irrelevant label...

**Comment by Benoit72-droid** [comment]
From the export page, the .asreview format is not available.
I have 4 choices: CSV (UTF-8), TSV(UTF-8), RIS, Excel

**Reply by Rensvandeschoot** [reply]
![image](https://github.com/asreview/asreview/assets/36502709/aa0e3366-72a8-4293-9e2b-d972f467f7fa)

**Reply by Benoit72-droid** [reply]
Thank you
